packages/react-ui-core/src/AutoSuggestField/AutoSuggestField.js

```diff
@@ -0,0 +1,205 @@
+    document.removeEventListener('keydown', this.onKeyDown)
+  }
+
+  @autobind
+  onKeyDown(event) {
+    const { suggestions } = this.props
+    const { visible } = this.state
+
+    switch (event.keyCode) {
+      case ESCAPE:
+        event.preventDefault()
+        this.reset()
+        break
+      case ENTER:
+        if (!visible || !suggestions || suggestions.length < 1) {
+          event.preventDefault()
+          this.handleOnSubmit(this.state.value)
+        }
+        break
+      default:
+    }
+  }
+
+  @autobind
+  handleInput(event) {
+    const value = event.target.value
+    this.setState({ value })
+    this.onInput(value)
+  }
+
+  @autobind
+  handleOnSubmit() {
+    this.props.onSubmit(this.state.value)
+    this.setState({ visible: false })
+  }
+
+  @autobind
+  handleSuggestionSelection(value) {
+    const {
+      onSubmit,
+      onSelection,
+      submitOnSelection,
+      onInput,
+    } = this.props
+
+    this.setState({
+      value,
+      visible: false,
+    })
+
+    onInput(value)
+    onSelection(value)
+    if (submitOnSelection) onSubmit(value)
+  }
+
+  @autobind
+  handleVisibilityChange(visible) {
+    this.setState({
+      visible,
+    })
+  }
+
+  @autobind
+  reset() {
+    this.setState({ value: '', visible: false })
+    this.props.onAfterReset()
+  }
+
+  renderButton(button, handleClick, dataTid, props) {
+    if (!button) return undefined
+
+    return createElement(...parseArgs(button, Button, {
+      'data-tid': dataTid,
+      onClick: handleClick,
+      ...props,
+    }))
+  }
+
+  render() {
+    const {
+      theme,
+      resetButton,
+      submitButton,
+      placeholder,
+      suggestions,
+      className,
+      onSubmit, // eslint-disable-line no-unused-vars
+      onSelectionHover,
+      onSelection, // eslint-disable-line no-unused-vars
+      suggestionValueSelector,
+      value: valueProps,
+      onAfterReset,
+      submitOnSelection,
+      ...props
+    } = this.props
+
+    const {
+      value,
+      visible,
+    } = this.state
+
+    return (
+      <div
+        className={classnames(
+          theme.AutoSuggestField,
+          className
+        )}
+        data-tid="autosuggest-field"
+      >
+
+        <Dropdown
+          anchorField={InputAnchor}
+          value={value}
+          onChange={this.handleInput}
+          visible={visible}
+          onVisibilityChange={this.handleVisibilityChange}
+          placeholder={placeholder}
+          toggle={false}
+          {...props}
+        >
+          <Menu
+            options={suggestions}
+            onSelection={this.handleSuggestionSelection}
+            handleSelectionHover={onSelectionHover}
+            optionValueSelector={suggestionValueSelector}
+          />
+        </Dropdown>
+
+        {this.renderButton(submitButton, this.handleOnSubmit, 'autosuggest-submit-button', ...props)}
+        {this.renderButton(resetButton, this.reset, 'autosuggest-reset-button', ...props)}
```

Comment:
How about something more like this:
```
this.renderButton(resetButton, { ...props, onClick: this.reset, 'data-tid': 'autosuggest-reset-button' })
```

This way you only have 2 arguments to worry about.

```
renderButton(button, props) {
  if (!button) return null

  return createElement(...parseArgs(button, Button, props))
}
```

Please also be careful about passing props like this here.  This means that both buttons get the same treatment as the `Dropdown`.